CMakeLists.txt

```diff
@@ -257,128 +274,61 @@ endif()
 #   ENSMALLEN_INCLUDE_DIR - include directory for ensmallen
 #   STB_IMAGE_INCLUDE_DIR - include directory for STB image library
 #   MATHJAX_ROOT - root of MathJax installation
-find_package(Armadillo "${ARMADILLO_VERSION}" REQUIRED)
 
+# Remove support for older 32bit system, if we need to have 32bit system
+# Since we are cross compiling, We will download it into the deps folder,
+find_package(Armadillo "${ARMADILLO_VERSION}")
+if(NOT ARMADILLO_FOUND)
+  get_deps(https://freefr.dl.sourceforge.net/project/arma/armadillo-10.3.0.tar.xz armadillo armadillo-10.3.0.tar.xz)
+  find_package(Armadillo)
+endif()
 # Include directories for the previous dependencies.
-set(MLPACK_INCLUDE_DIRS ${MLPACK_INCLUDE_DIRS} ${ARMADILLO_INCLUDE_DIRS})
-set(MLPACK_LIBRARIES ${MLPACK_LIBRARIES} ${ARMADILLO_LIBRARIES})
+set(MLPACK_INCLUDE_DIRS ${MLPACK_INCLUDE_DIRS} ${ARMADILLO_INCLUDE_DIR})
+
+# We do not have to link with Armadillo if we are cross compiling
+if(CMAKE_CROSSCOMPILING)
+  get_deps(https://github.com/xianyi/OpenBLAS/releases/download/v0.3.13/OpenBLAS-0.3.13.tar.gz OpenBLAS OpenBLAS-0.3.13.tar.gz)
+ 
+  ## I am open to this solution find_library() but only if it works :D.
+  # find_library(GFORTRAN_LIBRARY NAMES gfortran PATHS ${CMAKE_SYSROOT}/usr/lib)
+
+  ## This solution is good only if you use buildroot toolchain, for other toolchains users should overwrite these
+  ## variables in all cases
+  set(GFORTRAN_LIBRARY "${CMAKE_SYSROOT}/usr/lib/libgfortran.a")
+  set(PTHREAD_LIBRARY "${CMAKE_SYSROOT}/usr/lib/libpthread.a")
+  message(STATUS "GFORTRAN_LIBRARY ${GFORTRAN_LIBRARY}")
+  message(STATUS "PTHREAD_LIBRARY ${PTHREAD_LIBRARY}")
```

Comment:
Instead of `set()`, you can probably use `find_library()` to set `GFORTRAN_LIBRARY` and `PTHREAD_LIBRARY`.  This way, if a user specifies either variable, CMake won't actually search for it; but if they don't specify the variable, then CMake will search all of its usual directories (including `CMAKE_SYSROOT` I think).

Reply:
I tried to use `find_library ()` it never worked, I am happy to find an example to show me if it will work one day.

Reply:
It shouldn't be too hard: https://cmake.org/cmake/help/latest/command/find_library.html

I think that something like this could work:

```
find_library(GFORTRAN_LIBRARY NAMES libgfortran.a gfortran.a libgfortran.dll.a gfortran libgfortran)
```

This may be helpful too: https://cmake.org/pipermail/cmake/2010-December/041326.html

I haven't tried that command directly, but the way it should work is that if the user specifies `GFORTRAN_LIBRARY` on the command line, `find_library()` will do nothing.  In this case, I would expect that if you set `CMAKE_SYSROOT` to the root of the buildroot installation, it *should* be able to find gfortran... it might take a little debugging and playing with it to get it right, but it definitely should work. :+1:

Reply:
I will give it a try thanks, I wrote something similar even with specifying the path, I did not try to add all cases of gfortran, I will add this one and see what will happen :+1: